src/huggingface_hub/hf_api.py

```diff
@@ -1888,6 +1893,118 @@ def repo_info(
             files_metadata=files_metadata,
         )
 
+    @validate_hf_hub_args
+    def repo_exists(
+        self,
+        repo_id: str,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> bool:
+        """
+        Checks if a repository exists on the Hugging Face Hub.
+
+        Args:
+            repo_id (`str`):
+                A namespace (user or an organization) and a repo name separated
+                by a `/`.
+            repo_type (`str`, *optional*):
+                Set to `"dataset"` or `"space"` if getting repository info from a dataset or a space,
+                `None` or `"model"` if getting repository info from a model. Default is `None`.
+            token (`bool` or `str`, *optional*):
+                A valid authentication token (see https://huggingface.co/settings/token).
+                If `None` or `True` and machine is logged in (through `huggingface-cli login`
+                or [`~huggingface_hub.login`]), token will be retrieved from the cache.
+                If `False`, token is not sent in the request header.
+
+        Returns:
+            True if the repository exists, False otherwise.
+
+        <Tip>
+
+        Raises the following error:
+
+            - [`~utils.RepositoryNotFoundError`]
+                If the repository to download from cannot be found. This may be because it doesn't exist,
+                or because it is set to `private` and you do not have access.
+
+        Example:
+
+        repo_exists("huggingface/transformers")
+        True
+
+        repo_exists("huggingface/not-a-repo")
+        False
+
+        </Tip>
+        """
+        try:
+            repo_info(repo_id, repo_type=repo_type, token=token)
```

Comment:
```suggestion
            self.repo_info(repo_id, repo_type=repo_type, token=token)
```

Use the instance method instead of the global one. The instance method can be setup with difference token, user-agent, metadata,...

Reply:
After performing this change, my tests are failing. I need to fully understand why... hope to have some time on the weekend.

Reply:
> After performing this change, my tests are failing.

I think it's most probably due to pointing to 2 different endpoints (staging and production). By default, tests are run against the staging environment ("https://hub-ci.huggingface.co") but in the tests you've written, you expect to test against the production environment ("https://huggingface.co"). When you do `repo_info(...)` it uses the root-level defined method which is by default pointing to the production. But when you do `self.repo_info(...)` it uses the instance-defined method [which is pointing to the staging](https://github.com/huggingface/huggingface_hub/blob/b9901bf71839850c10ea42162c69aecabc96a46b/tests/test_hf_api.py#L120). I hope this short explanation (+ https://github.com/huggingface/huggingface_hub/pull/1591#discussion_r1296776435) makes it a bit clearer. Please let me know if it's not the case :grimacing:

Reply:
Ah, right... that makes sense...

I only have two of the asserts failing:

```python
def test_repo_exists(self):
   self.assertTrue(self._api.repo_exists(self.repo_id, token=False))  # private repo
```
and
```python
def test_file_exists(self):
   self.assertTrue(self._api.file_exists(self.repo_id, "file.txt"))
```

As soon as I have these figured out, we should be good to go.

Reply:
Yes, that was exactly it... thanks for the pointer!